src/sbvr-api/errors.ts

```diff
@@ -10,5 +10,237 @@ export class InternalRequestError extends TypedError {}
 
 export class TranslationError extends TypedError {}
 export class ParsingError extends TypedError {}
-export class BadRequestError extends TypedError {}
-export class ForbiddenError extends TypedError {}
+
+export class HttpError extends TypedError {
+	constructor(public status: number, error: string | Error = '') {
+		super(error);
+	}
+}
+
+export class BadRequestError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(400, error);
+	}
+}
+
+export class UnauthorizedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(401, error);
+	}
+}
+
+export class ForbiddenError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(403, error);
+	}
+}
+
+export class NotFoundError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(404, error);
+	}
+}
+
+export class MethodNotAllowedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(405, error);
+	}
+}
+
+export class NotAcceptableError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(406, error);
+	}
+}
+
+export class ProxyAuthenticationRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(407, error);
+	}
+}
+
+export class RequestTimeoutError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(408, error);
+	}
+}
+
+export class ConflictError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(409, error);
+	}
+}
+
+export class GoneError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(410, error);
+	}
+}
+
+export class LengthRequiredError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(411, error);
+	}
+}
+
+export class PreconditionFailedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(412, error);
+	}
+}
+
+export class PayloadTooLargeError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(413, error);
+	}
+}
+
+export class URITooLongError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(414, error);
+	}
+}
+
+export class UnsupportedMediaTypeError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(415, error);
+	}
+}
+
+export class RequestedRangeNotSatisfiableError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(416, error);
+	}
+}
+
+export class ExpectationFailedError extends HttpError {
+	constructor(error: string | Error = '') {
+		super(417, error);
+	}
+}
+
+export class IMATeapotError extends HttpError {
```

Comment:
You skipped `402 Payment Required` I think you can skip this one as well safely :P

Reply:
Got the errors from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
and missed 402 by accident (the html was different so I failed to extract that part).
Will add 402 and remove the teapot :+1:

Reply:
Oh 402 is future reserved and not meant to be used atm!
Anyway just give me a note and can remove the 402